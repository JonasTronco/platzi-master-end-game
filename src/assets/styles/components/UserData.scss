@import '../main.scss';

.user-data {
  width: 100%;
  margin-top: $space_x6;
  padding-left: $space_x6;
  display: flex;
  flex-direction: column;
  &__title {
    margin-top: $space_x6;
  }
  &__container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    max-width: 600px;
    height: 100%;
    justify-content: space-between;
  }
  &__element {
    display: flex;
    flex-direction: column;
    width: 210px;
    margin: $space_x3 0;
    &__title {
      font-family: $title_font;
    }
  }
}
