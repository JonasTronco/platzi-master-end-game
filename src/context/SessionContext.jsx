import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
