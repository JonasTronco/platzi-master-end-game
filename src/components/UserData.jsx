import PropTypes from 'prop-types';
import '../assets/styles/components/UserData.scss';

function UserData({ user }) {
  return (
    <div className="user-data">
      <h2 className="user-data__title"> User Information</h2>
      <div className="user-data__container">
        <div className="user-data__element">
          <strong className="user-data__element__title">Name</strong>
          <span className="user-data__element_content">{user.firstName}</span>
        </div>
        <div className="user-data__element">
          <strong className="user-data__element__title">Last Name</strong>
          <span className="user-data__element_content">{user.lastName}</span>
        </div>
        <div className="user-data__element">
          <strong className="user-data__element__title">Document ID</strong>
          <span className="user-data__element_content">{user.documentID}</span>
        </div>
      </div>
      <h2 className="user-data__title"> Contact</h2>
      <div className="user-data__container">
        <div className="user-data__element">
          <strong className="user-data__element__title">email</strong>
          <span className="user-data__element_content">{user.email}</span>
        </div>
        <div className="user-data__element">
          <strong className="user-data__element__title">Phone Number</strong>
          <span className="user-data__element_content">
            {user.contactNumber}
          </span>
        </div>
      </div>
    </div>
  );
}

UserData.propTypes = {
  user: PropTypes.object.isRequired,
};

export default UserData;
