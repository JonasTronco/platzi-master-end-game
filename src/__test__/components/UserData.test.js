import UserData from '../../components/UserData';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('UserData Component', () => {
  test('Should render UserData component', () => {
    const component = create(
      <SessionProvider>
        <UserData user={{}} />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
