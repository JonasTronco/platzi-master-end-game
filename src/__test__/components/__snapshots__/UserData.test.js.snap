// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserData Component Should render UserData component 1`] = `
<div
  className="user-data"
>
  <h2
    className="user-data__title"
  >
     User Information
  </h2>
  <div
    className="user-data__container"
  >
    <div
      className="user-data__element"
    >
      <strong
        className="user-data__element__title"
      >
        Name
      </strong>
      <span
        className="user-data__element_content"
      />
    </div>
    <div
      className="user-data__element"
    >
      <strong
        className="user-data__element__title"
      >
        Last Name
      </strong>
      <span
        className="user-data__element_content"
      />
    </div>
    <div
      className="user-data__element"
    >
      <strong
        className="user-data__element__title"
      >
        Document ID
      </strong>
      <span
        className="user-data__element_content"
      />
    </div>
  </div>
  <h2
    className="user-data__title"
  >
     Contact
  </h2>
  <div
    className="user-data__container"
  >
    <div
      className="user-data__element"
    >
      <strong
        className="user-data__element__title"
      >
        email
      </strong>
      <span
        className="user-data__element_content"
      />
    </div>
    <div
      className="user-data__element"
    >
      <strong
        className="user-data__element__title"
      >
        Phone Number
      </strong>
      <span
        className="user-data__element_content"
      />
    </div>
  </div>
</div>
`;
